app/src/main/java/com/android/sample/model/recipe/MealDBRecipesRepository.kt

```diff
@@ -0,0 +1,238 @@
+package com.android.sample.model.recipe
+
+import com.android.sample.resources.C.Tag.MAXIMUM_RECIPES_TO_FETCH_MEAL_DB
+import com.android.sample.resources.C.Tag.MAX_NB_OF_INGREDIENTS_IN_A_RECIPE_MEAL_DB
+import com.android.sample.resources.C.Tag.MEAL_DB_USER_AGENT
+import com.android.sample.resources.C.Tag.MEAL_DB_USER_AGENT_VALUE
+import java.io.IOException
+import okhttp3.Call
+import okhttp3.Callback
+import okhttp3.OkHttpClient
+import okhttp3.Request
+import okhttp3.Response
+import org.json.JSONException
+import org.json.JSONObject
+
+/**
+ * Repository for fetching recipes from the MealDB API.
+ *
+ * @property client The OkHttpClient used to make network requests.
+ */
+class MealDBRecipesRepository(private val client: OkHttpClient) : RecipesRepository {
+
+  private val mealDBUrl = "https://www.themealdb.com/api/json/v1/1/"
```

Comment:
Please move your constant to the resources/c file for better organization and consistency within the project.

Reply:
Done